docfiles/script.html

```diff
@@ -2,162 +2,196 @@
+                    </script>
+        </div>
+        <div class="script-bookend">
+            <div style="text-align: center;">
+
+                    <div class="ui container horizontal mini link list">
+                        <!-- <a class="ui item " href="https://makecode.com/contact " target="_blank " rel="noopener ">Contact Us</a> -->
+                        <a class="ui item " href="https://makecode.com/privacy " target="_blank " rel="noopener ">Privacy &amp; Cookies</a>
+                        <a class="ui item " href="https://makecode.com/termsofuse " target="_blank " rel="noopener "> Terms Of Use</a>
+                        <a class="ui item " href="https://makecode.com/trademarks " target="_blank " rel="noopener ">Trademarks</a>
+                        <div class="ui item ">© 2019 Microsoft</div>
+                    </div>
+                </div>
+        </div>
+
+        @body@
+    </div>
+
+    <script>
+        var editorEmbedURL = "/@versionsuff@#sandbox:@id@";
+        var gameEmbedURL = "/@versionsuff@--run?fullscreen=1&nofooter=1&id=@id@";
+        var showCodeButton = document.getElementById("show-code-button");
+        var frame = document.getElementById("embed-frame");
+
+        var isGame = true;
+
+        showCodeButton.addEventListener("click", function() {
+            // toggle game vs. code view
+            if (isGame) {
+                frame.setAttribute("src", editorEmbedURL);
+                showCodeButton.textContent = "Show Game";
+            }
+            else {
+                frame.setAttribute("src", gameEmbedURL);
+                showCodeButton.textContent = "Show Code";
+            }
+            isGame = !isGame;
+        });
+
+        var shareLinkIsApproved = false;
+
+        window.pxtTargetConfigPromise
+            .then(function(targetConfig) {
+                var approvedShareLinks = targetConfig && targetConfig.shareLinks && targetConfig.shareLinks.approved;
+                if (approvedShareLinks && approvedShareLinks.indexOf("@id@" >= 0)) {
+                    // This share link has been allow listed; remove cookie banner and allow sending of tick events
+                    shareLinkIsApproved = true;
+                    var abuseMessage = document.querySelector("#abuse-message");
+                    if (abuseMessage)
+                        abuseMessage.remove();
+                }
+            });
+
+        window.addEventListener('message', function (ev) {
+            if (!shareLinkIsApproved)
+                return;
+            var d = ev.data;
+            if (d.type == "messagepacket" && d.channel == "tick-event" && d.data) {
+                var unpackedData = "";
+                for (var i = 0; i < d.data.length; ++i)
+                    unpackedData += String.fromCharCode(d.data[i]);
+                try {
+                    const data = JSON.parse(unpackedData);
+                    window.pxtTickEvent(
+                        'simulator.user.tick',
```

Comment:
I think we need the share link id in the tick as well. It is easy to summarize if every tick had the id.

Reply:
Added the tick under shareurl; here's what the tick looks like when it reaches pxtTickEvent right now:

![image](https://user-images.githubusercontent.com/5615930/99628711-8f1c9300-29eb-11eb-8b31-f3a0f5dc5606.png)

I had `game` as a field intended to summarize off of in the asphodel, as that way it can keep the same query even if we need to update a sample link with a bug fix / updated extension / etc. Still useful to have the shareurl in general though as a backup / some info even if the tick from the user is an empty array.